dll/win32/shell32/CShellLink.cpp

```diff
@@ -3083,3 +3079,60 @@ HRESULT WINAPI IShellLink_ConstructFromFile(IShellFolder * psf, LPCITEMIDLIST pi
 
     return IShellLink_ConstructFromPath(path, riid, ppv);
 }
+
+HICON CShellLink::CreateShortcutIcon(LPCWSTR wszIconPath, INT IconIndex)
+{
+    const INT cx = GetSystemMetrics(SM_CXICON), cy = GetSystemMetrics(SM_CYICON);
+    const COLORREF crMask = GetSysColor(COLOR_3DFACE);
+    HDC hDC;
+    HIMAGELIST himl = ImageList_Create(cx, cy, ILC_COLOR32 | ILC_MASK, 1, 1);
+    HICON hIcon = NULL, hNewIcon = NULL;
+    HICON hShortcut = LoadIconW(shell32_hInstance, MAKEINTRESOURCE(IDI_SHELL_SHORTCUT));
+
+    ::ExtractIconExW(wszIconPath, IconIndex, &hIcon, NULL, 1);
+    if (!hIcon || !hShortcut || !himl)
+        goto cleanup;
```

Comment:
The code at `cleanup` calls DestroyIcon and ImageList_Destroy on these handles. Is that okay if they're null?

Reply:
Added checks.